sky/packages/sky/lib/src/fn3/framework.dart

```diff
@@ -712,11 +719,11 @@ abstract class BuildableElement<T extends Widget> extends Element<T> {
   /// components dirty during event handlers before the frame begins, not during
   /// the build itself.
   void markNeedsBuild() {
-    assert(!_debugStateLocked);
     assert(_debugLifecycleState == _ElementLifecycle.mounted);
     if (_dirty)
       return;
     _dirty = true;
+    assert(!_debugInBuildScope);
```

Comment:
> Can you elaborate on why we want markNeedsBuild()/setState() to be ok to run during that component's own build() function?

It made the code in ScrollableWidgetList easier to write.  It checks some invariants at the top of build.  If those invariants are false, it calls some general-purpose functions to restore them.  The general-purpose functions call setState because they don't know they're being called during build.  Without this change, they'll hit the _debugInBuildScope check.  With this change, they work correctly.